### PR TITLE
Upgrade language manager

### PR DESCRIPTION
**Pull request by eugeniosant**
Upgrade of language manager (with cache wrapper) to facilitate insert of a DataGrid layer (realized by Infinispan connector into components project).

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/com/agiletec/aps/system/services/lang/LangManager.java [hunk 1]
```diff
@@ -33,74 +34,51 @@
...
+ *
  * @author M.Diana - E.Santoboni
  */
 public class LangManager extends AbstractService implements ILangManager {
 
 	private static final Logger _logger = LoggerFactory.getLogger(LangManager.class);
-	
+
+	private Map<String, Lang> _assignableLangs;
+
```
[the remark this answers is not in the document]
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/aps/system/services/lang/LangManager.java]: I'll fixt in every class I modify

File: engine/src/main/java/com/agiletec/aps/system/services/lang/LangManager.java [hunk 2]
```diff
@@ -33,74 +34,51 @@
...
 		try {
-			InputStream is = this.getClass().getResourceAsStream("ISO_639 -1_langs.xml");
+			InputStream is = this.getClass().getResourceAsStream("ISO_639-1_langs.xml");
 			String xmlConfig = FileTextReader.getText(is);
 			LangDOM langDom = new LangDOM(xmlConfig);
 			List<Lang> tempList = langDom.getLangs();
-			this._assignableLangs = new HashMap<>();
-			for (int i=0; i<tempList.size(); i++){
+			this._assignableLangs = new HashMap<String, Lang>();
+			for (int i = 0; i < tempList.size(); i++) {
```
[the remark this answers is not in the document]
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/aps/system/services/lang/LangManager.java]: Ok, I'll change the "old style" for statement

File: engine/src/main/java/com/agiletec/aps/system/services/lang/cache/LangManagerCacheWrapper.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		cache.put(LANG_CODES_CACHE_NAME, langCodes);
+	}
+
+	@Override
+	public List<Lang> getLangs() {
+		List<Lang> langs = new ArrayList<Lang>();
+		Cache cache = this.getCache();
+		List<String> codes = (List<String>) this.get(cache, LANG_CODES_CACHE_NAME, List.class);
+		if (null != codes) {
+			for (int i = 0; i < codes.size(); i++) {
```
[the remark this answers is not in the document]
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/aps/system/services/lang/cache/LangManagerCacheWrapper.java]: I'll change in every class I modify

File: engine/src/test/java/com/agiletec/aps/system/services/lang/LangManagerIntegrationTest.java [hunk 4]
```diff
@@ -1,111 +1,112 @@
...
+		String langCode = lang.getCode();
+		String langDescr = lang.getDescr();
+		assertEquals(langCode, "it");
+		assertEquals(langDescr, "Italiano");
+	}
+
+	public void testGetLangs() throws ApsSystemException {
+		List<Lang> langs = this._langManager.getLangs();
+		assertEquals(2, langs.size());
+		for (int i = 0; i < langs.size(); i++) {
```
[the remark this answers is not in the document]
Comment (eugeniosant) [reply engine/src/test/java/com/agiletec/aps/system/services/lang/LangManagerIntegrationTest.java]: Ok

File: engine/src/main/java/com/agiletec/aps/system/services/lang/cache/LangManagerCacheWrapper.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+	}
+
+	private List<Lang> loadSystemLangs(String xmlConfig) throws ApsSystemException {
+		LangDOM langDom = new LangDOM(xmlConfig);
+		List<Lang> tempList = langDom.getLangs();
+		List<Lang> langList = new ArrayList<Lang>(tempList.size());
+		for (int i = 0; i < tempList.size(); i++) {
+			Lang lang = (Lang) tempList.get(i);
+			langList.add(lang);
+		}
```
[the remark this answers is not in the document]
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/aps/system/services/lang/cache/LangManagerCacheWrapper.java]: Useless code, I remove it

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Pull request merged**